src/DataTypes/Serializations/SerializationUUID.cpp

```diff
@@ -136,23 +137,31 @@ void SerializationUUID::deserializeBinary(IColumn & column, ReadBuffer & istr, c
 void SerializationUUID::serializeBinaryBulk(const IColumn & column, WriteBuffer & ostr, size_t offset, size_t limit) const
 {
     const typename ColumnVector<UUID>::Container & x = typeid_cast<const ColumnVector<UUID> &>(column).getData();
-
-    size_t size = x.size();
-
-    if (limit == 0 || offset + limit > size)
+    if (const size_t size = x.size(); limit == 0 || offset + limit > size)
         limit = size - offset;
 
-    if (limit)
+    if (limit == 0)
+        return;
+
+    if constexpr (std::endian::native == std::endian::big)
+    {
+        std::ranges::for_each(
```

Comment:
I probably wrote it elsewhere: `std::ranges` is rarely used throughout the codebase and most devs are unfamiliar with it. It would be cool to rewrite the code into a non-ranges version if the logic is simple enough. If not, it's also okay as "only" the BE path is affected.

Reply:
For sure, it wouldn't be particularly complex that way either. I'd probably go about it this way:
```
std::for_each(x.cbegin() + offset, x.cbegin() + offset + limit, [&ostr](const auto & uuid) { writeBinaryLittleEndian(uuid, ostr); });
```

The difference isn't night and day, although I think the ranges version reads quite nicely. Nonetheless, would you prefer that I submit another PR around these?